lib/src/io_utils.dart

```diff
@@ -118,40 +120,102 @@ final RegExp partOfRegexp = RegExp('part of ');
     'as Dartdoc 1.0.0')
 final RegExp newLinePartOfRegexp = RegExp('\npart of ');
 
-/// Best used with Future<void>.
-class MultiFutureTracker<T> {
-  /// Approximate maximum number of simultaneous active Futures.
-  final int parallel;
+typedef TaskQueueClosure<T> = Future<T> Function();
+
+class _TaskQueueItem<T> {
+  _TaskQueueItem(this._closure, this._completer, {this.onComplete});
+
+  final TaskQueueClosure<T> _closure;
+  final Completer<T> _completer;
+  void Function() onComplete;
+
+  Future<void> run() async {
+    try {
+      final result = await _closure();
+      if (result != null) {
+        _completer.complete(result);
+      } else {
+        _completer.complete(null);
+      }
+      await Future<void>.microtask(() {});
+    } catch (e) {
+      _completer.completeError(e);
+    } finally {
+      onComplete?.call();
+    }
+  }
+}
 
-  final Set<Future<T>> _trackedFutures = {};
+/// A task queue of Futures to be completed in parallel, throttling
+/// the number of simultaneous tasks.
+///
+/// The tasks return results of type T.
+class TaskQueue<T> {
+  /// Creates a task queue with a maximum number of simultaneous jobs.
+  /// The [maxJobs] parameter defaults to the number of CPU cores on the
+  /// system.
+  TaskQueue({int maxJobs})
+      : maxJobs = maxJobs ?? io.Platform.numberOfProcessors;
```

Comment:
+1; great feature.

Reply:
Make the tools run really fast on my workstation too!